### PR TITLE
Add gradient test for Embedding

### PR DESCRIPTION
**Pull request by t-ae**
Related issue: #402
Add gradient test for `Embedding`.

### CODE REVIEW

File: Tests/TensorFlowTests/LayerTests.swift [hunk 1]
```diff
@@ -756,6 +756,19 @@ final class LayerTests: XCTestCase {
         let expected = Tensor<Float>([[[0.4, 0.3], [0.2, 0.1]], [[0.2, 0.1],[0.2, 0.1]]])
         XCTAssertEqual(output, expected)
     }
+    
+    func testEmbeddingGradient() {
+        let layer = Embedding<Float>(vocabularySize: 4, embeddingSize: 5)
```
Comment (Shashi456) [inline Tests/TensorFlowTests/LayerTests.swift]: Add a Tensorflow Python equivalent, afaik our functionalities do match.

Comment (t-ae) [reply Tests/TensorFlowTests/LayerTests.swift]: Added equivalent code.
Keras has `Embedding` layer but it takes one-hot vectors as input.
So I used `tf.gather` instead.

Comment (Shashi456) [reply Tests/TensorFlowTests/LayerTests.swift]: Should we also update our implementation to support one hot o.O?

Comment (t-ae) [reply Tests/TensorFlowTests/LayerTests.swift]: @jon-tow 
Thank you for clarification. Taking one-hot vectors was my misunderstanding.

I found that `gather` and `Embedding` give different gradients even though these outputs are same.
`swift-apis`'s `Embedding` is `gather`-based, so it doesn't match to `tf.keras.layer.Embedding`.
I'll look into it.

```python
def gather():
    indices = tf.convert_to_tensor([0, 1, 2, 1, 2, 2], dtype=tf.int32)
    embedding = tf.convert_to_tensor([
        [0.4, 0.2, 0.1, 0.0, 0.1],
        [0.9, 0.7, 0.5, 0.2, 0.1],
        [0.2, 0.4, 0.6, 0.3, 0.2],
        [0.1, 0.3, 0.8, 0.3, 0.3]])
    with tf.GradientTape() as t:
        t.watch(embedding)
        output = tf.gather(embedding, indices)
        y = tf.reduce_sum(output)
    grad_slice = t.gradient(y, embedding) # IndexedSlice
    grad = tf.zeros_like(embedding).numpy()
    for index in grad_slice.indices:
        grad[index] += grad_slice.values[index].numpy()
    print(output)
    print(grad)

def embedding():
    indices = tf.convert_to_tensor([0, 1, 2, 1, 2, 2], dtype=tf.int32)
    embeddings = tf.constant([
        [0.4, 0.2, 0.1, 0.0, 0.1],
        [0.9, 0.7, 0.5, 0.2, 0.1],
        [0.2, 0.4, 0.6, 0.3, 0.2],
        [0.1, 0.3, 0.8, 0.3, 0.3]])
    layer = tf.keras.layers.Embedding(4, 5, weights=[embeddings])

    with tf.GradientTape() as t:
        t.watch(layer.weights)
        output = layer(indices)
        y = tf.reduce_mean(output)
    grad_slice = t.gradient(y, layer.weights)[0] # IndexedSlice
    grad = tf.zeros_like(embeddings).numpy()
    for index in grad_slice.indices:
        grad[index] += grad_slice.values[index].numpy()
    print(output)
    print(grad)

gather()
embedding()
```

Comment (t-ae) [reply Tests/TensorFlowTests/LayerTests.swift]: @jon-tow 
Thank you for your clarification.
I updated python code to use `tf.keras.layer.Embedding`.

**Comment by rxwei** [comment]
How about applying #499 going forward?

**Comment by t-ae** [comment]
Updated.
